astropy/io/fits/tests/test_header.py

```diff
@@ -180,24 +180,24 @@ def test_assign_boolean(self):
         h = fits.Header()
         h.update('FOO', True)
         h.update('BAR', False)
-        assert h['FOO'] == True
-        assert h['BAR'] == False
+        assert h['FOO']
+        assert h['BAR'] is False
         assert h.ascard['FOO'].cardimage == fooimg
         assert h.ascard['BAR'].cardimage == barimg
 
         h = fits.Header()
         h.update('FOO', np.bool_(True))
         h.update('BAR', np.bool_(False))
-        assert h['FOO'] == True
-        assert h['BAR'] == False
+        assert h['FOO']
+        assert h['BAR'] is False
```

Comment:
Here, I'm not sure what the actual intent was...  But testing it, it would seem that the idea is that any semi-boolean input gets converted to a proper `Bool`, so again `assert h['FOO'] is True`